### ISSUE TITLE
Operator is stateless, so is non-parametric layer

### ISSUE THREAD

**Issue by tonyyang-svail** [PaddlePaddle/tape#23]
A paddle fluid operator is stateless. It only performs computation, doesn't store any intermediate value in its instance. Consider `c = mul(a, b)` and `e = mul(c, d)`, the first `mul` and second `mul` can be the same operator instance.

We can think of a neural net as a stack of layers. A layer may or may not contain parameters.

A non-parametric layer e.g. `softmax`, `activation`, `fill_constant`, is also stateless. Hence a non-parametric layer could be written as a c++ function. On the other hand, a parametric layer, e.g. `fc`, `conv`, should be instantiated explicitly. Because the layer owns its parameters, and therefore stateful.
